server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/service/ReprocessingAllTask.java

```diff
@@ -0,0 +1,104 @@
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.service;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import javax.mail.MessagingException;
+
+import org.apache.james.mailrepository.api.MailRepositoryStore;
+import org.apache.james.task.Task;
+import org.apache.james.task.TaskExecutionDetails;
+
+import com.github.fge.lambdas.Throwing;
+
+public class ReprocessingAllTask implements Task {
+
+    public static final String TYPE = "reprocessingAllTask";
+
+    public class AdditionalInformation implements TaskExecutionDetails.AdditionalInformation {
+        private final Supplier<Long> countSupplier;
+        private final long initialCount;
+
+        public AdditionalInformation(Supplier<Long> countSupplier) {
+            this.countSupplier = countSupplier;
+            this.initialCount = countSupplier.get();
+        }
+
+        public String getTargetQueue() {
+            return targetQueue;
+        }
+
+        public Optional<String> getTargetProcessor() {
+            return targetProcessor;
+        }
+
+        public String getRepositoryUrl() {
+            return repositoryUrl;
+        }
+
+        public long getRemainingCount() {
+            return countSupplier.get();
+        }
+
+        public long getInitialCount() {
+            return initialCount;
+        }
+    }
+
+    private final ReprocessingService reprocessingService;
+    private final String repositoryUrl;
+    private final String targetQueue;
+    private final Optional<String> targetProcessor;
+    private final AdditionalInformation additionalInformation;
+
+    public ReprocessingAllTask(ReprocessingService reprocessingService, MailRepositoryStoreService mailRepositoryStoreService,
+                               String repositoryUrl, String targetQueue, Optional<String> targetProcessor) {
+        this.reprocessingService = reprocessingService;
+        this.repositoryUrl = repositoryUrl;
+        this.targetQueue = targetQueue;
+        this.targetProcessor = targetProcessor;
+        this.additionalInformation = new AdditionalInformation(
+            Throwing.supplier(() -> mailRepositoryStoreService.size(repositoryUrl).orElse(0L)));
```

Comment:
it looks like a bad idea. Having a DTO datastructure calling a service is dangerous IMO. We should rather update the status with a listener if needed.

Reply:
Gonna get it a try, but I smell that it gonna be `diff hungry`

Reply:
Do we need that information right now ? Maybe we can skip that field for now.